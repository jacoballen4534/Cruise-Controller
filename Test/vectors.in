On Off Resume Set QuickAccel QuickDecel Accel Brake Speed
# DO NOT modify first line to a comment will be used for header
###########################################################
################### Car Driving in STATE OFF ############## 
###########################################################
false false false false false false 0.0 0.0 0.0
false false false false false false 0.0 0.0 0.0
false false false false false false 51.234 0.0 0.0
false false false false false false 51.234 0.0 2.0937
false false false false false false 51.234 0.0 4.1862
false false false false false false 51.234 0.0 6.2772
false false false false false false 51.234 0.0 8.3661
false false false false false false 89.687 0.0 10.453
false false false false false false 89.687 0.0 13.051
false false false false false false 89.687 0.0 15.644
false false false false false false 89.687 0.0 18.233
false false false false false false 89.687 0.0 20.816
false false false false false false 89.687 0.0 23.393
false false false false false false 89.687 0.0 25.962
false false false false false false 89.687 0.0 28.524
false false false false false false 89.687 0.0 31.078
false false false false false false 89.687 0.0 33.623
# TEST 18:In OFF: Press all signals except ON, Off. No affect.
false false true true true true 0.0 0.0 36.158
###########################################################
################### NEXT STATE transitions ################
###########################################################
# TEST 19:Set On to true will go to On State,Cruise Speed will become current speed
true false false false false false 0.0 0.0 36.049
# TEST 20 :Stays in ON, current Speed<cruise Speed so will use Throttle to Accel
false false false false false false 0.0 0.0 35.94
# Test 21 : Use Brake -> Should go to Standby(4)
false false false false false false 0.0 10.0 36.0649
# Test 22: Brake and Resume, Brake should have priority due to strong abort
false false true false false false 0.0 10.0 36.0649
# Test 23:IN Standby: No resume Accel pressed should stay in Standby
false false false false false false 10.0 0.0 36.0649
# Test 24:IN Standby: Resume Accel pressed should go to Disable
false false true false false false 10.0 0.0 36.0649
# Test 25:IN Disable: SpeedLegal and No Accel pressed should go to ON
false false false false false false 0.0 0.0 36.0649
# Test 26:IN On: Accel pressed should go to Disable
false false false false false false 10.0 0.0 36.0649
# Test 27:IN Disable: Brake pressed should go to STDBY
false false false false false false 0.0 10.0 36.0649
# Test 28:IN STDBY: Resume , No brake/accel and legalSpeed should go to ON
false false true false false false 0.0 0.0 36.0649
# Test 29:IN ON: Off , press brake should go to OFF
false true false false false false 0.0 10.0 36.0649
#######################################################################
################### Cruise Speed Management in STATE OFF ##############
#######################################################################
# Test 30:IN OFF: Resume no impact (accel now drives throttle)
false false true false false false 10.0 0.0 36.0649
# Test 31:IN OFF: Set cant change cruiseSpeed in On
false false false true false false 0.0 0.0 36.0649
# Test 32:IN OFF: (Changing speed has no impact on CruiseSpeed/Throttle in State Off)
false false false false false false 0.0 0.0 35.06000
# Test 33:IN OFF: QuickAccel has no impact on CruiseSpeed
false false false false true false 0.0 0.0 35.06000
# Test 34:IN OFF: QuickDecel has no impact on CruiseSpeed
false false false false false true 0.0 0.0 35.06000
# Test 35:IN OFF: cruiseSpeed will be set to Vehicle Speed when going to On 
true false false false false false 0.0 0.0 35.06000
######################################################################
################### Cruise Speed Management in State On ##############
######################################################################
# Test 36:In ON: QuickDecel should reduce speed and stick to limit
false false false false false true 0.0 0.0 35.06000
# Test 37:In ON: QuickDecel should reduce again
false false false false false true 0.0 0.0 32.56001
# Test 38:In ON: QuickDecel should cap at SPEEDMIN
false false false false false true 0.0 0.0 32.56001
# Test 39:In ON: Set, cruiseSpeed should cap at SPEEDMAX and go to DISABLE
false false false true false false 0.0 0.0 200.00000
# Test 40:In DISABLE: Set, should set CruiseSpeed and go to ON(no aceel and legal speed)
false false false true false false 0.0 0.0 149.00000
# Test 41:In ON: QuickAccel should cap at SPEEDMAX (Curr Speeed= SPEEDMAX To avoid throttleCalcs)
false false false false true false 0.0 0.0 150.00000
# Test 42:In ON: Set, and QuickAccel.Set is first priority, others not executed
false false false true true false 0.0 0.0 100.00000
# Test 43:In ON: QuickAccel and QuickDecel.(Net affect zero)
false false false false true true 0.0 0.0 100.00000
# Test 44: In ON: Go back to OFF (All other signals dont matter on is strong abort)
true true true true true true 0.0 10.0 99.00000
###########################################################
################### Car Driving in STATE On ###############
###########################################################
# Test 45: In OFF: Go to ON (CruiseSpeed set to Vehicle Speed, Throttle only uses Kp*error)
true false false false false false 0.0 0.0 100.00000
# Test 46: In ON: throttle cmd now works
false false false false false false 0.0 0.0 98.00000
# Test 47: In ON: KI term increases
false false false false false false 0.0 0.0 99.00000
# Test 48: In ON: KI term increases again
false false false false false false 0.0 0.0 99.00000
# Test 49: In ON: KI term should be frozen
false false false false false false 0.0 0.0 100.00000
# Test 50: In ON: KI term remains frozen
false false false false false false 0.0 0.0 100.00000
# Test 51: In ON: Goes to Off and nothing changes
false true false false false false 0.0 0.0 100.00000
# Test 52: In OFF: Goes to On and Integral term reset to 0
true false false false false false 0.0 0.0 100.00000
# Test 53: In ON: ThrottleCMD Saturates
false false false false false false 0.0 0.0 94.00000
################ MAKE SURE NO BLANK LINES IN FILE ########################