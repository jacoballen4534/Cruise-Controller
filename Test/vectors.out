CruiseSpeed ThrottleCmd CruiseState
# DO NOT modify first line to a comment will be used for header
###########################################################
################### Car Driving in STATE OFF ############## 
###########################################################
0.000000 0.000000 0
0.000000 0.000000 0
0.000000 51.234001 0
0.000000 51.234001 0
0.000000 51.234001 0
0.000000 51.234001 0
0.000000 51.234001 0
0.000000 89.686996 0
0.000000 89.686996 0
0.000000 89.686996 0
0.000000 89.686996 0
0.000000 89.686996 0
0.000000 89.686996 0
0.000000 89.686996 0
0.000000 89.686996 0
0.000000 89.686996 0
0.000000 89.686996 0
0.000000 0.000000 0
###########################################################
################### NEXT STATE transitions ################
###########################################################
# TEST 19: 
36.049000 0.000000 1
# TEST 20: 
36.049000 0.938827 1
# TEST 21:
36.049000 0.000000 2
# TEST 22:
36.049000 0.000000 2
# Test 23:
36.049000 10.000000 2
# Test 24:
36.049000 10.000000 3
# Test 25:
36.049000 0.000000 1
# Test 26:
36.049000 10.000000 3
# Test 27:
36.049000 0.000000 2
# Test 28:
36.049000 0.000000 1
# Test 29:
36.049000 0.000000 0
######################################################################
################### Cruise Speed Management in State OFF ##############
######################################################################
# Test 30:
36.049000 10.000000 0
# Test 31:
36.049000 0.000000 0
# Test 32:
36.049000 0.000000 0
# Test 33:
36.049000 0.000000 0
# Test 34:
36.049000 0.000000 0
# Test 35:
35.060001 0.000000 1
######################################################################
################### Cruise Speed Management in State On ##############
######################################################################
# Test 36:
32.560001 0.000000 1
# Test 37:
30.060001 0.000000 1
# Test 38:
30.000000 0.000000 1
# Test 39:
150.000000 0.000000 3
# Test 40:
149.000000 0.000000 1
# Test 41:
150.000000 0.000000 1
# Test 42:
100.000000 0.000000 1
# Test 43:
100.000000 0.000000 1
# Test 44:
100.000000 0.000000 0
###########################################################
################### Car Driving in STATE Off->On ##########
###########################################################
# Test 45:
100.000000 0.000000 1
# Test 46:
100.000000 17.226000 1
# Test 47: In OFF: throttle cmd now works
100.000000 9.613000 1
# Test 48: In ON: KI term increases again
100.000000 10.113000 1
# Test 49: In ON: KI term increases again
100.000000 2.000000 1
# Test 50: In ON: KI term increases again
100.000000 2.000000 1
# Test 51: In ON: Goes to off 
100.000000 0.000000 0
# Test 52: In OFF: Goes to on
100.000000 0.000000 1
# Test 53: In ON: 
100.000000 45.000000 1
# MAKE SURE NO BLANK LINES IN FILE
